include/operators/binary_arithmetic.hpp

```diff
@@ -156,6 +156,14 @@ BigInt BigInt::operator*(const BigInt& num) const {
     BigInt product;
     if (abs(*this) <= FLOOR_SQRT_LLONG_MAX and abs(num) <= FLOOR_SQRT_LLONG_MAX)
         product = std::stoll(this->value) * std::stoll(num.value);
+    else if (is_power_of_10(this->value)){ // if LHS is a power of 10 do optimised operation 
+        product = num;
+        add_trailing_zeroes(product.value, get_power_of_10(this->value));
```

Comment:
Instead of adding trailing zeroes equal to the power of 10, use the following (it's much more efficient):
```C++
product.value.append(this->value.begin() + 1, this->value.end());
```